lineapy/graph_reader/artifact_collection.py

```diff
@@ -98,10 +110,19 @@ def __init__(
                 art._session_id, []
             ) + [art]
 
+        # Check we use cache in an isolated session, which is meaningless
+        for session, arts in artifacts_by_session.items():
+            if all([art.name not in artifact_names for art in arts]):
+                raise ValueError(
+                    f"{art.name} is not in any sessions that contain {', '.join(artifact_names)}"
```

Comment:
I think we should give the user some error message hint on how to avoid this. Perhaps something like, "Try removing the following artifacts from input_artifact: [art_1, art_2] ... "